test/mustache_test.rb

```diff
@@ -496,6 +496,24 @@ def test_sections_returning_lambdas_get_called_with_text
     assert_equal 1, view.calls
   end
 
+  def test_sections_returning_lambdas_get_called_dynamically_with_text
+    view = Mustache.new
+    view.template       = '{{name}}'
+    view[:name]         = lambda { '{{dynamic_name}}' }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use the new lambda literal syntax `-> {...}`.](https://www.codacy.com/app/rokusu/mustache/file/2000951366/issues/source?bid=2903613&fileBranchId=3121922#l502)